src/modules/facility-fixed-fee/dto/get-facility-fixed-fee-response.dto.ts

```diff
@@ -0,0 +1,100 @@
+import { ApiProperty } from '@nestjs/swagger';
+import { EXAMPLES, PROPERTIES } from '@ukef/constants';
+import { DateOnlyString } from '@ukef/helpers';
+
+export type GetFacilityFixedFeeResponse = GetFacilityFixedFeeResponseItem[];
+export class GetFacilityFixedFeeResponseItem {
+  @ApiProperty({
+    description: 'The identifier of the facility.',
+    example: EXAMPLES.FACILITY_ID,
+    minLength: 10,
+    maxLength: 10,
+  })
+  readonly facilityIdentifier: string;
+
+  @ApiProperty({
+    description: 'The identifier of the portfolio.',
+    example: PROPERTIES.GLOBAL.portfolioIdentifier,
+  })
+  readonly portfolioIdentifier: string;
+
+  @ApiProperty({
+    description: 'The fixed amount to be billed to the client. If a flat amount is charged, this field is required.',
+    example: EXAMPLES.DEAL_OR_FACILITY_VALUE,
+  })
+  readonly amount: number;
+
+  @ApiProperty({
+    description: 'The effective date of this accruing/Fixed fee schedule.',
+    type: Date,
+    format: 'date',
+    example: '2023-03-24',
+  })
+  readonly effectiveDate: DateOnlyString;
+
+  @ApiProperty({
+    description: 'The expiration date of this accruing/Fixed fee schedule.',
+    type: Date,
+    format: 'date',
+    example: '2023-03-24',
+  })
+  readonly expirationDate: DateOnlyString;
+
+  @ApiProperty({
+    description: 'Date the next fee bill is due for this fee schedule.',
+    type: Date,
+    format: 'date',
+    example: '2023-03-24',
+  })
+  readonly nextDueDate: DateOnlyString;
+
+  @ApiProperty({
+    description: 'End date for the current accrual period. This date can be different than the Next Due Date.',
+    type: Date,
+    format: 'date',
+    example: '2023-03-24',
+  })
+  readonly nextAccrueToDate: DateOnlyString;
+
+  @ApiProperty({
+    description: 'Segment identifier from income exposure table. 2 alphanumeric characters.',
+    maxLength: 2,
+    minLength: 2,
+  })
+  readonly period: string;
+
+  @ApiProperty({
+    description: 'Description of accruing/Fixed fee which is be entered by user.',
+  })
+  readonly description: string;
+
+  @ApiProperty({
+    description:
+      'The currency of Facility Fee, defined in the Currency Definition Table under Systems Administration of Servicing. For example, USD for United States Dollar.',
+    example: EXAMPLES.CURRENCY,
+    minLength: 0,
+    maxLength: 3,
+  })
+  readonly currency: string;
+
+  @ApiProperty({
+    description: 'Defines the code for the role of the party in the Facility for which the fee is created.',
+    example: '500',
+    minLength: 0,
+    maxLength: 3,
+  })
+  readonly lenderTypeCode: string;
+
+  @ApiProperty({
+    description: 'Defines the code for the fee type of this schedule for reporting and GL purposes.',
+    minLength: 0,
+    maxLength: 3,
```

Comment:
Are there pre-defined values which ACBS will accept, if yes then enum will be ideal.

Reply:
I think we should get the entire ENUM list making a `GET` ACBS call, sending an invalid `ENUM` can throw errors mid-way during records creation.

Thoughts @avaitonis?

Reply:
It's not easy to get all possible values, especially for POST. 
  * Field incomeClass has all values in reference table T1105, but we need ACBS UI to access it. 
  * Alternative is GET /Lists/IncomeClass , but I noticed that these List get calls not always use expected reference table.
  
Even if ACBS has ~5-20 incomeClasses, we not necessary need to support all. Original possible values might be provided by BA.
  
Also not sure if we want to maintain long ENUMs.

In this specific case GET /Lists/IncomeClass returns 58 items in DEV and 56 in PROD.

Reply:
My suggestion is to merge this PR with current ENUM values and review our ENUM strategy later.

Reply:
Based on above, it makes sense to review ENUM later. Thanks @avaitonis